yarn-project/simulator/src/avm/avm_gas_cost.ts

```diff
@@ -0,0 +1,94 @@
+import { Opcode } from './serialization/instruction_serialization.js';
+
+/** Gas cost in L1, L2, and DA for a given instruction. */
+export type GasCost = {
+  l1Gas: number;
+  l2Gas: number;
+  daGas: number;
+};
+
+/** Gas cost of zero across all gas dimensions. */
+export const EmptyGasCost = {
+  l1Gas: 0,
+  l2Gas: 0,
+  daGas: 0,
+};
+
+/** Dimensions of gas usage: L1, L2, and DA */
+export const GasDimensions = ['l1Gas', 'l2Gas', 'daGas'] as const;
+
+/** Temporary default gas cost. We should eventually remove all usage of this variable in favor of actual gas for each opcode. */
+const TemporaryDefaultGasCost = { l1Gas: 0, l2Gas: 10, daGas: 0 };
+
+/** Gas costs for each instruction. */
+export const GasCosts: Record<Opcode, GasCost> = {
```

Comment:
I thought you wanted to do it per instruction and variable?

Reply:
I realized most instructions will have a fixed cost, and it's easier to see them in a single place. Instructions with a variable cost can just override `gasCost`. I'll work on one in a future PR to act as an example!